tools/cldr-code/src/main/java/org/unicode/cldr/util/CLDRFile.java

```diff
@@ -3339,6 +3343,42 @@ private List<String> getRawExtraPathsPrivate() {
         return toAddTo.stream().map(String::intern).collect(Collectors.toList());
     }
 
+    private void getLanguageExtraPaths(Set<String> toAddTo) {
+        Set<String> codes =
+                StandardCodes.make().getSurveyToolDisplayCodes(NameType.LANGUAGE.getNameName());
+        codes.remove(XMLSource.ROOT_ID);
```

Comment:
this is wrong, my mistake -- it actually modifies the set in StandardCodes! Need to make a copy of the set. We might want to prevent this kind of bug by making getSurveyToolDisplayCodes return an unmodifiable set, maybe using [Set.copyOf](https://docs.oracle.com/javase/10/docs/api/java/util/Set.html#copyOf(java.util.Collection)).

We currently have this, which doesn't work: `CLDRLanguageCodes = CldrUtility.protectCollection(CLDRLanguageCodes);`

Reply:
Fixed by my last commit -- we still should look into Set.copyOf as an alternative to protectCollection

Reply:
Important: In general, use ImmutableSet.copyOf instead of Set.copyOf. The latter changes the order in the set, which may be important (TreeSet or LinkedHashSets get messed up, and it doesn't hurt for HashSet).

Reply:
Also, we need to fix CldrUtility.protectCollection(CLDRLanguageCodes); — if it does't work then a lot of other things could go wrong.